docs/specs/meta-clients/web3wallet/decrypted-norifications.md

```diff
@@ -0,0 +1,27 @@
+# Decrypted Notification
+
+The below class introduces the general way of handling decprytped notifications received by a wallet. The implementation details differs on each platform.
+
+```typescript
+abstract class DecryptedNotification {
+    // decrypt encrypted request
+    public decryptMessage(params: { topic: string, encryptedMessage: string }): Promise<Message>;
+
+    // get notification metadata for a given request
+    public getNotificationMetadata(params: { topic: string, message: Message }): Promise<Notification>;
+}
+
+type Message = {
+    id: string,
+    jsonrpc: string, 
+    method: string, 
+    params: string, //Sign(proposal, request) params, Auth request params, Notify params
```

Comment:
Clients will use `method` to know how to parse the params. Maybe removing this comment would help?

Reply:
Want I meant is that we should have documented somewhere which tags could be recevied, namely: 
- Sign: `SESSION_PROPOSE(1100)`, `SESSION_REQUEST*(1108)`
- Auth: `AUTH_REQUEST(3000)`
- Notify: `NOTIFY_MESSAGE(4002)`

Reply:
All of these messages would be received by the client. The client would filter them out via `method`

Reply:
What I mean is that we had this specified in the technical brief. Hence, I think it would also be beneficent to have all those tags listed in the specs as well. If they are not documented how clients would know which methods to filter?

Reply:
Gotcha, adding tags

Reply:
Hmm, so that's why I wanted to include a tag in the notification payload, to know what message to filter out before decryption. Now clients have to first decrypt to know a method and decide whether to display a given notification.

If we had a tag in the payload, clients would filter out notifications by a tag and decrypt only those that are needed. It seems more efficient. Wdyt?

Reply:
Added here: https://github.com/WalletConnect/walletconnect-specs/pull/147/commits/d83fc21af042c411ba08dfa16c6526324314d0f8